astroquery/cadc/core.py

```diff
@@ -270,80 +308,158 @@ def get_data_urls(self, query_result, include_auxiliaries=False):
+            warnings.warn('verbose deprecated since 0.4.0')
+        tables = self.get_tables()
+        for t in tables:
+            if table == t.name:
+                return t
+
+    def exec_sync(self, query, maxrec=None, uploads=None, output_file=None):
+        """
+        Run a query and return the results or save them in a output_file
+
+        Parameters
+        ----------
+        query : str, mandatory
+            SQL to execute
+        maxrec : int
+            the maximum records to return. defaults to the service default
+        uploads:
+            Temporary tables to upload and run with the queries
+        output_file: str or file handler:
+            File to save the results to
 
-    def query_async(self, query):
+        Returns
+        -------
+        Results of running the query in (for now) votable format
+
+        Notes
+        -----
+        Support for other output formats (tsv, csv) to be added as soon
+        as they are available in pyvo.
         """
-        Runs a query and returns results in Table format
+        response = self.cadctap.search(query, language='ADQL',
+                                       uploads=uploads)
+        result = response.to_table()
+        if output_file:
+            if isinstance(output_file, str):
+                with open(output_file, 'bw') as f:
+                    f.write(result)
+                    return
+            else:
+                output_file.write(result)
+                return
+        return result
+
+    def create_async(self, query, maxrec=None, uploads=None):
+        """
+        Creates a TAP job to execute and returns it to the caller. The
+        caller then can start the execution and monitor the job.
+        Typical (no error handling) sequence of events:
+
+        job = create_async(query)
+        job = job.run().wait()
+        job.raise_if_error()
+        result = job.fetch_result()
+        job.delete() # optional
+
+        See ``pyvo.dal.tap`` for details about the ``AsyncTAPJob``
 
         Parameters
         ----------
-        query: str
-            query: query to run
+        query : str, mandatory
+            SQL to execute
+        maxrec : int
+            the maximum records to return. defaults to the service default
+        uploads:
+            Temporary tables to upload and run with the queries
+        output_file: str or file handler:
+            File to save the results to
 
         Returns
         -------
+        AsyncTAPJob
+            the query instance
 
-        results in `~astropy.table.Table` format
+        Notes
+        -----
+        Support for other output formats (tsv, csv) to be added as soon
+        as they are available in pyvo.
         """
-        return self.run_query(query, operation='sync')
+        return self.cadctap.submit_job(query, language='ADQL',
+                                       uploads=uploads)
 
+    @deprecated('0.4.0', 'Use axec_sync or create_async instead')
```

Comment:
Just noticed this now! I think this should be `exec_sync` instead of `axec_sync`